rfcs/NNNN-hot-elements.md

```diff
@@ -0,0 +1,81 @@
+---
+Status: Active
+Champions: rictic
+PR: [15](https://github.com/lit/rfcs/pull/15)
+---
+
+# Hot Elements
+
+A new Lit Labs package for supporting hot module replacement with custom elements.
+
+## Objective
+
+### Goals
+- Instantiating a new custom element, post HMR, should use the newest version of the constructor.
+- Provide an API for custom element classes so that they can add additional handling of HMR. For example, in LitElement we can patch and rerender existing elements in the page.
+  - This API should be implementable by a base class, so that in many cases no changes are needed to an element for it to achieve optimal HMR.
+- Support @web/dev-server-hmr specifically and the [esm-hmr](https://github.com/FredKSchott/esm-hmr) protocol more generally.
```

Comment:
is it worth specifying somewhere in this doc which popular tools we want to make sure we definitely support?

off the top of my head - web-dev-server, vite, webpack, storybook. AFAIK all follow the same protocol roughly.

but maybe its worth defining that we should test that at least these work with it? though integration tests might be a bit of a stretch i guess. just a thought